### PR TITLE
Issue055/add card overview

### PR DESCRIPTION
**Pull request by wisskirchenj**

### CODE REVIEW

File: frontend/src/feature/cards/components/CardItemScroller.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  if (pagePointer.value.isLast) {
+    done('empty');
+    return;
+  }
+  const cardResponse = await useCardsService().getCards(props.categoryId, props.titleFilter, pagePointer.value.current);
+  pagePointer.value.current = cardResponse.currentPage + 1;
+  pagePointer.value.isLast = cardResponse.isLast;
+  for (const cardItem of cardResponse.cards) {
+    if (items.value.find((item) => item.id === cardItem.id)) {
+      continue; // necessary, since method is called manually and automatically after filter change
```
Comment (alfabravo2013) [inline frontend/src/feature/cards/components/CardItemScroller.vue:45]: I guess no LinkedHashSet structure is available?

File: server/src/main/java/org/hyperskill/community/flashcards/card/CardService.java [hunk 2]
```diff
@@ -40,27 +39,31 @@ public class CardService {
...
-        );
-
-        var count = mongoTemplate.count(new Query(), category.name());
-        var result = mongoTemplate.aggregate(aggregation, category.name(), Document.class)
-                .getRawResults()
-                .getList("results", Document.class);
-        var cards = result.stream().map(converter::convert).toList();
+        var pageRequest = PageRequest.of(page, PAGE_SIZE, Sort.by("title"));
+        var query = createFilterQuery(titleFilter);
+        var count = mongoTemplate.count(query, category.name());
```
Comment (alfabravo2013) [inline server/src/main/java/org/hyperskill/community/flashcards/card/CardService.java:48]: the count query is now fixed, right?

**Review by alfabravo2013 — APPROVED** [review]

File: frontend/src/feature/cards/components/CardItemScroller.vue (hunk 1 above)
Comment (wisskirchenj) [reply frontend/src/feature/cards/components/CardItemScroller.vue:45]: I thought of using a Set - that exists in JS. But often those collection structures are not as precise as we know them form Java. And I was not sure how it works with the components for-loop in template area...

File: server/src/main/java/org/hyperskill/community/flashcards/card/CardService.java (hunk 2 above)
Comment (wisskirchenj) [reply server/src/main/java/org/hyperskill/community/flashcards/card/CardService.java:48]: yep. it works now as expected.
Though on still needs care. I need to keep a strict order:
1. create query with applied filters
2. count with this query
3. use query.with(pageRequest) as the latter modifies the query - and then count always gives just the page size ...

**Pull request merged**